## 1.5 有理点のなす群

In [1]:
def f(x)
  x**3 + 3 * x + 4
end

(0..6).each do |x|
  puts(format("(x, y**2) = (%d, %d)", x, f(x)))
end

(x, y**2) = (0, 4)
(x, y**2) = (1, 8)
(x, y**2) = (2, 18)
(x, y**2) = (3, 40)
(x, y**2) = (4, 80)
(x, y**2) = (5, 144)
(x, y**2) = (6, 238)


0..6

* $ 4 = (\pm 2)^2 $
    * $ (0, 2) $
    * $ (0, -2) = (0, 5) $
* $ 8 = 1 = (\pm 1)^2 $
    * $ (1, 1) $
    * $ (1, -1) = (1, 6) $
* $ 18 = 4 = (\pm 2)^2 $
    * $ (2, 2) $
    * $ (2, -2) = (2, 5) $
* ~~40 = 5~~
* ~~80 = 3~~
* $ 144 = 4 = (\pm 2)^2 $
    * $ (5, 2) $
    * $ (5, -2) = (5, 5) $
* 238 = 0
    * $ (6, 0) $

$ \Rightarrow $ 素体 $ F_7 $ 上の楕円曲線 $ y^2 = x^3 + 3x + 4 $ の群位数は 10

(無限遠点をカウント)

> 素体 $ F_p $ 上の楕円曲線 E の群位数 $ \#E $ は
> $$ p + 1 - 2 \sqrt[]{\mathstrut p} \leq \#E \leq p + 1 + 2 \sqrt[]{\mathstrut p} $$

$ F_7 $ 上の楕円曲線 $ y^2 = x^3 + 3x + 4 $ について考える。

In [2]:
E = 10
p = 7
p + 1 - 2 * Math.sqrt(p) <= E && E <= p + 1 + 2 * Math.sqrt(p)

true